### PR TITLE
Add show_all param to report list endpoint to always show all reports on a branch

### PR DESCRIPTION
**Pull request by r-ash**

### CODE REVIEW

File: tests/testthat/test-z-integration.R [hunk 1]
```diff
@@ -384,16 +411,18 @@ test_that("can get report parameters", {
 })
 
 test_that("can get report parameters with no commit ID", {
-  path <- orderly_prepare_orderly_git_example()
-  ## Checkout a branch with a report with parameters
-  orderly_git_checkout_branch("other", root = path[["local"]])
-  server <- start_test_server(path[["local"]])
+  path <- orderly_prepare_orderly_example("interactive", testing = TRUE,
```
Comment (r-ash) [inline tests/testthat/test-z-integration.R:414]: This is a change of behaviour which is maybe unexpected due to changes in `git_latest_commit`. Previously this was relying on the fact that `/reports/<name>/parameters` endpoint if called without a commit would use the currently checked out version of the repo. It now defaults to using master branch. We probably weren't relying on the previous default behaviour? What default behaviour do you think we should use? I can make some further changes if previous made more sense, or maybe even just not support calling this endpoint without a parameter for commit id.

File: R/git.R [hunk 2]
```diff
@@ -143,17 +143,16 @@ git_commits <- function(branch, root = NULL) {
   commits
 }
 
-git_latest_commit <- function(root = NULL) {
-  args <- c("log", "--pretty='%h'", "-n", "1")
+git_latest_commit <- function(branch = "master", root = NULL) {
```
Comment (richfitz) [inline R/git.R:146]: I think we need to change this to look up the default branch otherwise we're going to hit problems. Can you look to see how widespread this is, and we might be better off doing it as a different PR.

File: R/git.R [hunk 3]
```diff
@@ -143,17 +143,16 @@ git_commits <- function(branch, root = NULL) {
...
-get_reports <- function(branch, commit, root) {
-  if (identical(branch, "master") || is.null(branch)) {
-    ## Get all reports in commit if on master branch and default to latest
-    ## if commit is missing
+get_reports <- function(branch, commit, show_all, root) {
+  list_all <- show_all || identical(branch, "master") || is.null(branch)
+  if (list_all) {
     if (is.null(commit)) {
-      commit <- git_latest_commit(root = root)
+        commit <- git_latest_commit(branch, root = root)
```
Comment (richfitz) [inline R/git.R:155]:
```suggestion
      commit <- git_latest_commit(branch, root = root)
```

File: R/git.R [hunk 4]
```diff
@@ -143,17 +143,16 @@ git_commits <- function(branch, root = NULL) {
...
+  args <- c("log", "--pretty='%h'", "-n", "1", branch)
   git_run(args, root = root, check = TRUE)$output
 }
 
-get_reports <- function(branch, commit, root) {
-  if (identical(branch, "master") || is.null(branch)) {
-    ## Get all reports in commit if on master branch and default to latest
-    ## if commit is missing
+get_reports <- function(branch, commit, show_all, root) {
+  list_all <- show_all || identical(branch, "master") || is.null(branch)
```
Comment (richfitz) [inline R/git.R:152]: Also here for master vs default branch

**Review by richfitz — CHANGES_REQUESTED** [review]
Looks straightfoward enough - but there's something I noticed about the default branch that is going to bite us at some point as everyone moves from master to main (the default gh instructions use main now, and so both will be live for some time).  It might be better dealt with as a different PR though (also one trivial bit of noise in the diff)

File: tests/testthat/test-z-integration.R (hunk 1 above)
Comment (richfitz) [reply tests/testthat/test-z-integration.R:414]: I think we should use the new behaviour, and use the default branch - that way if someone has checked out a branch to experiment on the server, the API behaviour is consistent

**Comment by r-ash** [comment]
We also refer to master in `git_branches_no_merged` and `git_commits` so I think would be good to sort that default branch in a separate PR

**Comment by r-ash** [comment]
Here's a ticket https://mrc-ide.myjetbrains.com/youtrack/issue/VIMC-5772

**Review by EmmaLRussell — APPROVED** [review]
This works as expected - when I run OW with this branch and local demo, and use show_all I see global and minimal reports on 'other' branch, but not if show_all is omitted.
